### PR TITLE
fix: sorted update will cause list to flicker

### LINKED ISSUE #5702: infinity scroll has some bugs
https://user-images.githubusercontent.com/90852205/232527543-f1efd3e4-8b9e-44f7-b096-9bee4c4a622e.mov
- not scrolling on the recording 
- try scrolling on explore

### LINKED ISSUE #6235: Sort option triggers erratic behavior on big collections
### What happened?

Collection example: 
https://canary.kodadot.xyz/rmrk/collection/08C79124CAC59DC643-BOREDBARRO?collectionId=08C79124CAC59DC643-BOREDBARRO

https://canary.kodadot.xyz/ksm/collection/e63e926f7db5997e5c-STGR?collectionId=e63e926f7db5997e5c-STGR&sort=instance_ASC&page=2

https://github.com/kodadot/nft-gallery/assets/36627808/59e2b698-eb0c-408b-b569-4f5f6550c341

### Please reproduce in steps

Go to specified collection -> sort by any of the options -> scroll down, wait -> scroll down

### Expected Behavior

this should not happen :) 

### What browsers are you seeing the problem on?

Desktop Chrome, Desktop Firefox

### At which address did you encounter bug?

canary.kodadot.xyz

### Are you logged in?

No

### Which wallet you are using?

_No response_

### At which chain did you encounter bug?

_No response_

### Screenshots

^
 |

### Relevant log output

_No response_

### Payment link for reward

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #6833: Weird glitch on explorer when scrolling
Is this already solved in some PR or new?

https://github.com/kodadot/nft-gallery/assets/5887929/be0f2ee0-d0be-49cd-acd7-aeed259213f4

### PR DESCRIPTION
**Pull request by newraina**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #6833 
- [x] Closes #6235
- [x] Closes #5702 
- [ ] Requires deployment <snek/rubick/worker>

### Root cause 1
`query.sort` is an array that triggers the `watch`'s callback even if the content hasn't changed, but only the reference has changed (e.g. `[] --> []`), resulting in a `resetPage` when loading the previous or next page of data.

### How to fix
While watching for changes to `query.sort`, use deep comparison to determine if they are equivalent.

### Root cause 2:

https://github.com/kodadot/nft-gallery/blob/8388f5a52621faaa464480bc810a1b7e9b304971/components/items/ItemsGrid/useItemsGrid.ts#L85-L94

There is a logical inconsistency between `nfts.value` and `loadedPages.value`:

If the current page already exists in `loadedPages`, the `nfts` is not considered to need updating.
However, in components that consume `nfts`, when query parameters such as search/sort are updated, the value in `nfts` is assumed to be outdated, and `nfts` is cleared before the query is initiated. In this case, `loadedPages` is not cleared, so the `nfts` update is skipped when the query function is invoked.

### How to fix

1. `nfts` should be treated as read-only for external components that consume it, and its reset should be handled by `useFetchSearch` exposing a special function so that `useFetchSearch` can know whether `nfts` has been reset or the current query really has no data.

2. within `useFetchSearch`, whenever `nfts` is cleared, `loadedPages` is also cleared.

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer?target=1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce&usdamount=0&donation=true)

#### Community participation

- [x] [Are you at KodaDot Ecosystem Telegram?](https://t.me/kodadot_eco)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at 7a2a63b</samp>

Improved the NFT search functionality by clearing the previous results before fetching new ones. Modified the `ItemsGrid` component and the `useFetchSearch` composable to implement this feature.

### <samp>🤖 Generated by Copilot at 7a2a63b</samp>

> _To improve the NFTs display_
> _We added a function today_
> _It clears the old fetch_
> _With a simple `clearFetch`_
> _And refetches with a new query_

### CODE REVIEW

**Comment by yangwao** [comment]
nice combo! I think this is worth extra $!:)

**Comment by prury** [comment]
Explorer a bit less glitchy, but still glitchy: 

https://github.com/kodadot/nft-gallery/assets/36627808/a6e7357b-7a3e-4369-81b1-2d670a505370

Inside a collection, i still can't reach the end(try sorting by sequence): 

https://github.com/kodadot/nft-gallery/assets/36627808/b6055e45-034f-4502-92c1-cf2ce4d1473e

**Comment by newraina** [comment]
@prury you can try again.

I also found that the "scrolling problem after selecting sort" also occurs in the collection list, also fixed it in this PR:

https://github.com/kodadot/nft-gallery/assets/10708802/02b53cf1-b774-49cd-963f-9016744fcaf2

**Comment by prury** [comment]
> @prury you can try again.
> 
> I also found that the "scrolling problem after selecting sort" also occurs in the collection list, also fixed it in this PR:
> CleanShot.2023-09-04-g2gC2DF6.mp4

now it works, ty!

**Review by prury — APPROVED** [review]

File: components/items/ItemsGrid/useItemsGrid.ts [hunk 1]
```diff
@@ -113,7 +118,10 @@ export function useFetchSearch({
       () => route.query.owned,
       () => route.query.collections,
     ],
-    () => {
+    ([currentSort], [prevSort]) => {
```
Comment (Jarsen136) [inline components/items/ItemsGrid/useItemsGrid.ts:121]: how about `[...currentSort]` ?

**Review by Jarsen136 — CHANGES_REQUESTED** [review]
Something was broken.
image

**Comment by prury** [comment]
Thank you for catching it @Jarsen136

**Comment by newraina** [comment]
> @newraina nothing happens when filtering from a featured collection

@prury @Jarsen136 fixed.

**Review by prury — APPROVED** [review]

**Review by Jarsen136 — APPROVED** [review]

**Review by daiagi — APPROVED** [review]

**Comment by newraina** [comment]
> nice combo! I think this is worth extra $!:)

@yangwao Looking forward to the extra $ 👀

**Comment by yangwao** [comment]
pay 70 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $70 @ 4.04 USD/DOT ~ 17.327 $DOT
🧗 [1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce](https://polkadot.subscan.io/account/1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce)
🔗 [0x83a8035b41bda958afb90a1705d3a60ca93adfdec26892221c4611a4475f9193](https://polkadot.subscan.io/extrinsic/0x83a8035b41bda958afb90a1705d3a60ca93adfdec26892221c4611a4475f9193)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**